napari/plugins/tests/test_hookspecs.py

```diff
@@ -0,0 +1,87 @@
+import inspect
+import warnings
+
+import pytest
+from numpydoc.docscrape import FunctionDoc
+
+from napari.plugins import hookspecs
+
+# 1. we first create a hookspec decorator:
+#    ``hookspec = pluggy.HookspecMarker("napari")```
+# 2. when it decorates a function, that function object gets a new attribute
+#    called "napari_spec"
+# 3. that attribute is what makes them discoverable when you run
+#    ```plugin_manager.add_hookspecs(module)```
+#
+# here, we are using that attribute to discover all of our internal hookspecs
+# (in module ``napari.plugins.hookspecs``) so as to make sure that they conform
+# to our own internal rules about documentation and type annotations, etc...
+
+HOOKSPECS = []
+for name, func in vars(hookspecs).items():
+    if hasattr(func, 'napari_spec'):
+        HOOKSPECS.append((name, func))
+
+
+@pytest.mark.parametrize("name, func", HOOKSPECS)
+def test_hookspec_naming(name, func):
+    """All hookspecs should begin with napari_ (until we decide otherwise)"""
```

Comment:
... until pluggy releases a new version with your PR in it? I'd love to drop this requirement as soon as possible.

Reply:
That's sort of a separate (but not totally unrelated) issue.  This test just makes sure that all of _our_ hook specifications start with the name `napari` (it's a convention adopted by most users of pluggy),  My pluggy PR allows a developer to indicate that one of their functions meets the API of one of our hooks specifications without renaming _their_ function, for instance:

```python
@napari_hook_specification(specname='napari_get_reader')
def their_function(a, b):
    ...
```

I think it's a good convention.  If someday there is a package of hook specifications that are _not_ napari-specific (like a set of community agreed-upon APIs). then _those_ specifications should not start with the name `napari`... 

make sense?  thoughts?

Reply:
Makes sense to me